lib/active_record/connection_adapters/sqlserver/type/time_value_fractional.rb

```diff
@@ -39,7 +39,7 @@ def fractional_operator
           end
 
           def fractional_precision
-            0.00333
+            0.00333.to_d
```

Comment:
When doing literals, please use BigDecimal('0.00333') instead. Try to stay away from to_d, it has a slightly different behavior when used with Floats and the above is more accurate.